libs/agno/agno/storage/base.py

```diff
@@ -1,60 +1,74 @@
+    ):
+        if (
+            not agent_sessions_table_name
+            and not team_sessions_table_name
+            and not workflow_sessions_table_name
+            and not memory_table_name
+            and not learnings_table_name
+            and not eval_runs_table_name
+        ):
+            raise ValueError("At least one of the tables must be provided")
 
-    @mode.setter
-    def mode(self, value: Optional[Literal["agent", "team", "workflow"]]) -> None:
-        """Set the mode of the storage."""
-        self._mode = "agent" if value is None else value
+        self.agent_sessions_table_name = agent_sessions_table_name
+        self.team_sessions_table_name = team_sessions_table_name
+        self.workflow_sessions_table_name = workflow_sessions_table_name
+        self.memory_table_name = memory_table_name
+        self.learnings_table_name = learnings_table_name
+        self.eval_runs_table_name = eval_runs_table_name
 
-    @abstractmethod
-    def create(self) -> None:
-        raise NotImplementedError
+    # --- READ ---
 
     @abstractmethod
-    def read(self, session_id: str, user_id: Optional[str] = None) -> Optional[Session]:
+    def read_session(self, session_id: str) -> Optional[Session]:
         raise NotImplementedError
 
     @abstractmethod
-    def get_all_session_ids(self, user_id: Optional[str] = None, agent_id: Optional[str] = None) -> List[str]:
+    def get_all_session_ids(self, agent_id: Optional[str] = None) -> List[str]:
         raise NotImplementedError
 
     @abstractmethod
-    def get_all_sessions(self, user_id: Optional[str] = None, entity_id: Optional[str] = None) -> List[Session]:
+    def get_all_sessions(self, entity_id: Optional[str] = None) -> List[Session]:
         raise NotImplementedError
 
-    @abstractmethod
-    def get_recent_sessions(
-        self,
-        user_id: Optional[str] = None,
-        entity_id: Optional[str] = None,
-        limit: Optional[int] = 2,
-    ) -> List[Session]:
-        raise NotImplementedError
+    # --- WRITE ---
 
     @abstractmethod
-    def upsert(self, session: Session) -> Optional[Session]:
+    def create_table(self) -> None:
         raise NotImplementedError
 
     @abstractmethod
     def delete_session(self, session_id: Optional[str] = None):
         raise NotImplementedError
 
-    @abstractmethod
-    def drop(self) -> None:
-        raise NotImplementedError
+    # @abstractmethod
+    # def upsert(self, session: Session) -> Optional[Session]:
+    #     raise NotImplementedError
 
-    @abstractmethod
-    def upgrade_schema(self) -> None:
-        raise NotImplementedError
+    # --- UTILITIES ---
+
+    # @abstractmethod
+    # def create_table(self, table_name: str, schema: str) -> None:
+    #     raise NotImplementedError
+
+    # @abstractmethod
+    # def validate_table_schema(self, table_name: str, schema: str) -> None:
+    #     raise NotImplementedError
+
+    # @abstractmethod
+    # def upgrade_schema(self) -> None:
```

Comment:
We definitely need a `upgrade` which just handles everything. And we store migrations as files outside of the DB.